src/apps/explorer/pages/UserDetails.tsx

```diff
@@ -40,21 +44,48 @@ const TitleAddress = styled(RowWithCopyButton)`
 const UserDetails: React.FC = () => {
   const { address } = useParams<{ address: string }>()
   const networkId = useNetworkId() || undefined
+  const [resolvedAddress, setResolvedAddress] = useState<string | undefined | null>()
 
-  if (!isAddress(address)) {
+  useEffect(() => {
+    async function _resolveENS(): Promise<void> {
+      const _address = await resolveENS(address)
+      setResolvedAddress(_address)
+    }
+
+    if (isAddress(address)) {
+      setResolvedAddress(address)
+    } else if (isEns(address)) {
+      _resolveENS()
+    }
+  }, [address, networkId])
+
```

Comment:
Not sure it would work, so feel free to ignore my suggestion.

What about creating a hook for this: `useResolveEns(address?:string) -> string | undefined` ?

Reply:
I like it, I think it can work. Let me try...